cpg-library/src/main/java/de/fraunhofer/aisec/cpg/frontends/java/StatementAnalyzer.java

```diff
@@ -184,12 +184,11 @@ private ForEachStatement handleForEachStatement(Statement stmt) {
     de.fraunhofer.aisec.cpg.graph.statements.Statement iterable =
         lang.getExpressionHandler().handle(forEachStmt.getIterable());
 
-    if (!(var instanceof DeclarationStatement
-        && ((DeclarationStatement) var).isSingleDeclaration())) {
-      log.error("more or unknown decl in foreach");
-    } else {
-      statement.setVariable(((DeclarationStatement) var).getSingleDeclaration());
-    }
+    if (!(var instanceof DeclarationStatement))
+      log.error("Expected a DeclarationStatement but received: {}", var.getName());
+    else if (!((DeclarationStatement) var).isSingleDeclaration())
+      log.error("Expected a single DeclarationStatement.");
+    else statement.setVariable(var);
```

Comment:
While you are at it: Could you please rename `var` to `variable`, so that it does not clash with the keyword var of later Java version / Kotlin.